<?xml version="1.0"?>
<launch>

  <!--Robot description and parameter files -->
    <arg name="calibrate" default="true" doc="if inventory file is generated"/>
    <arg name="robot_ip" default="10.0.0.2" doc="UR5's IP" />
    <arg name="kinematics_config" default="/home/ratatouille/my_robot_calibration.yaml" />
    <arg name="rviz_config" default="$(find ur5e_moveit_config)/launch/moveit_ratatouille_autonomous_dispensing.rviz" />
    <arg name="realsense_camera_type" default="d435" />

    <!-- Robot bringup -->
    <!-- <include file="$(find ur_robot_driver)/launch/ur5e_bringup.launch">
    <arg name="kinematics_config" value="$(arg kinematics_config)"/>
    <arg name="robot_ip" value="$(arg robot_ip)"/>
  </include> -->

    <!-- UR5e Moveit -->
    <include file="$(find ur5e_moveit_config)/launch/ur5e_moveit_planning_execution.launch"></include>

    <!-- Sensors -->
    <include file="$(find sensor_interface)/launch/start_sensors.launch" />

    <!-- Robot Gripper controller -->
    <node pkg="robotiq_urcap_control" type="robotic_urcap_ctrl_py3_wrapper.py" name="robotiq_urcap_control">
      <!--<param name="publish_frequency" type="double" value="$(arg tf_pub_rate)" />
    <param name="tf_prefix" value="$(arg tf_prefix)" />  -->
    </node>

    <!-- Rviz launch -->
    <include file="$(find ur5e_moveit_config)/launch/moveit_rviz.launch">
      <arg name="rviz_config" value="$(arg rviz_config)" />
    </include>

    <group if="$(arg calibrate)">
      <!-- RealSense Camera -->
      <include file="$(find realsense2_camera)/launch/rs_rgbd.launch">
        <arg name="device_type" value="$(arg realsense_camera_type)" />
        <arg name="enable_depth" value="true" />
      </include>

      <!-- April Tag Tracking -->
      <include file="$(find ar_track_alvar)/launch/realsense_rgb_track.launch" />

      <!-- Pose estimation -->
      <node pkg="ratatouille_pose_estimation" type="publish_marker_poses.py" name="ratatouille_pose_estimation" />

      <!-- Ingredient identification -->
      <node pkg="ingredient_validation" type="validate_ingredient_service.py" name="ingredient_validation" />
    </group>
</launch>